packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
-    onCountChange={onCountChange}
-  />
-);
+  min = 0,
+  max = 100,
+  onCountChange,
+}) => {
+  const [counterValue, setCounterValue] = useState(value);
+  const handleChange = (value: number) => setCounterValue(value);
+
+  return (
+    <Counter
+      ref={ref}
+      label={label}
+      labelId={labelId}
+      error={error}
+      id={id}
+      description={description}
+      value={counterValue}
+      data-testid={dataTestId}
+      min={min}
+      max={max}
+      onCountChange={onCountChange ?? handleChange}
+    />
+  );
+};
 
 test('forwards ref', () => {
   const ref = createRef<HTMLInputElement>();
-  const { container } = render(counterMock({ ref, ...requiredAttributes }));
-  const counter = container.querySelector('input');
 
-  expect(ref.current).toBe(counter);
+  render(<Counter value={5} onCountChange={jest.fn()} ref={ref} />);
+
+  const input = screen.getByRole<HTMLInputElement>('textbox');
+
+  expect(ref.current).toBe(input);
 });
 
 test('renders a counter as an input tag', () => {
-  const { container } = render(counterMock(requiredAttributes));
+  render(<CounterMock value={5} />);
 
-  expect(container.querySelector('input')).toBeInTheDocument();
+  expect(screen.getByLabelText(/Label/i)).toBeInTheDocument();
 });
 
 test('renders a counter with matched label', () => {
-  const { queryByLabelText } = render(counterMock({ label: 'Test Label', ...requiredAttributes }));
+  render(<CounterMock value={5} label="Test Label" />);
 
   // This one checks for matching id and htmlFor
-  expect(queryByLabelText('Test Label')).toBeInTheDocument();
+  expect(screen.getByLabelText(/test label/i)).toBeInTheDocument();
 });
 
 test('create unique ids if not provided', () => {
-  const { queryByTestId } = render(
+  render(
     <>
-      {counterMock({ dataTestId: 'item1', label: 'Test Label', ...requiredAttributes })}
-      {counterMock({ dataTestId: 'item2', label: 'Test Label', ...requiredAttributes })}
+      {<CounterMock value={5} dataTestId="item1" label="Test Label" />}
+      {<CounterMock value={5} dataTestId="item2" label="Test Label" />}
     </>,
   );
 
-  const item1 = queryByTestId('item1') as HTMLInputElement;
-  const item2 = queryByTestId('item2') as HTMLInputElement;
+  const item1 = screen.getByTestId<HTMLInputElement>('item1');
+  const item2 = screen.getByTestId<HTMLInputElement>('item2');
 
   expect(item1).toBeDefined();
   expect(item2).toBeDefined();
   expect(item1.id).not.toBe(item2.id);
 });
 
 test('respects provided id', () => {
-  const { container } = render(counterMock({ id: 'test', label: 'Test Label', ...requiredAttributes }));
-  const counter = container.querySelector('#test') as HTMLInputElement;
+  const providedIdValue = 'test';
+
+  render(<CounterMock value={5} id={providedIdValue} label="Test Label" />);
 
-  expect(counter.id).toBe('test');
+  const counterId = screen.getByLabelText<HTMLInputElement>(/test label/i).id;
+
+  expect(counterId).toBe(providedIdValue);
 });
```

Comment:
Prefer 

```ts
const counter = screen.getByLabelText(/test label/i);

expect(counter).toHaveAttribute('id', providedIdValue);
```

If there's an error, this will show a better error message.

<img width="636" alt="image" src="https://user-images.githubusercontent.com/4542735/145308990-dd978488-ad04-4781-99fe-326fa089c954.png">

vs

<img width="641" alt="image" src="https://user-images.githubusercontent.com/4542735/145309039-47f84351-e6b8-4526-b3ae-31ae02e78446.png">

Reply:
Noted, I haven't thought about it in this way.